django/db/models/sql/query.py

```diff
@@ -419,6 +420,10 @@ def get_aggregation(self, using, aggregate_exprs):
                 getattr(self.annotations[ref], "subquery", False)
                 for ref in aggregate.get_refs()
             )
+            refs_window |= any(
+                getattr(self.annotations[ref], "contains_over_clause", True)
```

Comment:
2 things made me stop and think at this line:

1. The `True` default seems wrong here? 🤔 We're assuming that if the attr doesn't exist then it is a window expression?
2. All expressions have the attribute `contains_over_clause` defined due to the [property](https://github.com/django/django/blob/e8252fc4451ee2c05779b28e74417174f0a2eeef/django/db/models/expressions.py#L247) on `BaseExpression` so does that mean that the `getattr()` is superfluous here?

Reply:
@shangxiao here's an example of the most recent regression due to the lack of usage of `getattr` in ticket-34745